Source/Core/Core/Analytics.cpp

```diff
@@ -0,0 +1,237 @@
+#include <cinttypes>
+#include <memory>
+#include <mutex>
+#include <random>
+#include <string>
+#include <mbedtls/sha1.h>
+
+#if defined(_WIN32)
+#include <windows.h>
+#elif defined(__APPLE__)
+#include <CoreServices/CoreServices.h>
+#endif
+
+#include "Common/Analytics.h"
+#include "Common/Common.h"
+#include "Common/CommonTypes.h"
+#include "Common/CPUDetect.h"
+#include "Common/StringUtil.h"
+#include "Core/Analytics.h"
+#include "Core/ConfigManager.h"
+#include "Core/Movie.h"
+#include "Core/NetPlayProto.h"
+#include "VideoCommon/VideoConfig.h"
+
+namespace
+{
+constexpr const char* ANALYTICS_ENDPOINT = "https://analytics.dolphin-emu.org/report";
+}  // namespace
+
+std::mutex DolphinAnalytics::s_instance_mutex;
+std::shared_ptr<DolphinAnalytics> DolphinAnalytics::s_instance;
+
+DolphinAnalytics::DolphinAnalytics()
+{
+	ReloadConfig();
+	MakeBaseBuilder();
+}
+
+std::shared_ptr<DolphinAnalytics> DolphinAnalytics::Instance()
+{
+	std::lock_guard<std::mutex> lk(s_instance_mutex);
+	if (!s_instance)
+	{
+		s_instance.reset(new DolphinAnalytics());
+	}
+	return s_instance;
+}
+
+void DolphinAnalytics::ReloadConfig()
+{
+	std::lock_guard<std::mutex> lk(m_reporter_mutex);
+
+	// Install the HTTP backend if analytics support is enabled.
+	std::unique_ptr<Common::AnalyticsReportingBackend> new_backend;
+	if (SConfig::GetInstance().m_analytics_enabled)
+	{
+		new_backend = std::make_unique<Common::HttpAnalyticsBackend>(ANALYTICS_ENDPOINT);
+	}
+	m_reporter.SetBackend(std::move(new_backend));
+
+	// Load the unique ID or generate it if needed.
+	m_unique_id = SConfig::GetInstance().m_analytics_id;
+	if (m_unique_id.empty())
+	{
+		GenerateNewIdentity();
+	}
+}
+
+void DolphinAnalytics::GenerateNewIdentity()
+{
+	std::random_device rd;
+	u64 id = (static_cast<u64>(rd()) << 32) | rd();
+	m_unique_id = StringFromFormat("%016" PRIx64, id);
+
+	// Save the new id in the configuration.
+	SConfig::GetInstance().m_analytics_id = m_unique_id;
+	SConfig::GetInstance().SaveSettings();
+}
+
+std::string DolphinAnalytics::MakeUniqueId(const std::string& data)
+{
+	u8 digest[20];
+	std::string input = m_unique_id + data;
+	mbedtls_sha1(reinterpret_cast<const u8*>(input.c_str()), input.size(),
+	             digest);
+
+	// Convert to hex string and truncate to 64 bits.
+	std::string out;
+	for (int i = 0; i < 8; ++i)
+	{
+		out += StringFromFormat("%02hhx", digest[i]);
+	}
+	return out;
+}
+
+void DolphinAnalytics::ReportDolphinStart(const std::string& ui_type)
+{
+	Common::AnalyticsReportBuilder builder(m_base_builder);
+	builder.AddData("type", "dolphin-start");
+	builder.AddData("ui-type", ui_type);
+	builder.AddData("id", MakeUniqueId("game-start"));
```

Comment:
Tiny nitpick, this should be "dolphin-start" not "game-start".

Doesn't affect implementation just self-documentation.